### DISCUSSION TITLE
How to exit "sudo" mode?

### DISCUSSION THREAD

**Discussion by sos4nt** [community/community#23100] [New to GitHub]
<p>GitHub has a so-called <a href="https://help.github.com/articles/sudo-mode/" rel="nofollow noopener">sudo mode</a> which I think is a great security feature. However, after entering my password the sudo mode persists for quite a while: (emphasis added)</p>
<blockquote>
<p>After you’ve performed a sudo-protected action, you’ll only be asked to re-authenticate again after <strong>a few hours</strong> of inactivity.</p>
</blockquote>
<p>How can I manually <em>exit</em> that mode?</p>

**Comment by ThirdEyeRose** [comment]
<p>Hi <a class="mention" href="/sos4nt">@sos4nt</a>,</p>
<p>It is currently not possible to manually exit sudo as such. If you can tell me more about how this feature would help you, I’d be happy to relay that feedback to the appropriate teams.</p>
<p>Thanks!</p>

**Comment by sos4nt** [comment]
<p>Hi <a class="mention" href="/that-pat">@that-pat</a>,</p>
<p>having to enter my password adds an extra layer of security for <em>sudo-protected</em> actions. And it’s quite convenient that I can temporarily remove that restriction to perform multiple such actions in succession. But removing the restriction for several <em>hours</em> just doesn’t feel right. (e.g. the timeout in Unix sudo defaults to 5 minutes)</p>

**Comment by ThirdEyeRose** [comment, marked as answer]
<p>Hi <a class="mention" href="/sos4nt">@sos4nt</a>,</p>
<p>Thanks for the additional information! We’re always working to improve GitHub and the GitHub Community Forum, and we consider every suggestion we receive. I’ve logged your feature request in our internal feature request list. Though I can’t guarantee anything or share a timeline for this, I have shared your suggestions regarding sudo mode with the appropriate teams for consideration.</p>
<p>Please let me know if you have any other questions.</p>
<p>Thanks!</p>

**Comment by breyn407** [comment]
<p>How do I disable this “feature?”  Looks like the infamous Windows “Are you sure?” dialog.</p>
<p>It is great if you want that enabled, but <strong>super annoying</strong> when you do not.</p>
<p>Also, it would be great if when new “features” like this are added, that they include an OFF switch by default.  Getting tired of the “we know better than you” security nanny posture.</p>

**Comment by queenofcorgis** [comment]
<p>Hi there <a class="mention" href="/breyn407">@breyn407</a> and welcome :wave:t3:</p>
<p>Are you looking to disable this feature after you’ve authorized it or prevent it from popping up entirely?</p>

**Reply by LongossNoah** [reply]
I want to prevent it from popping up entirely , how do I go about it, Thanking you in anticipation of a positive response

**Reply by queenofcorgis** [reply]
@LongossNoah This is not a configurable option, but a requirement to keep your account safe so you will not be able to disable it.

**Reply by breyn407** [reply]
Thankds @queenofcorgis.  How did I authorize this?  AFAIK, I did no such thing.  I do not see an option to turn it off, so I am unclear how one would have a choice to authorize or decline.  Thanks for your reply. :)

Oh, also, I saw your reply to @LongossNoah.  If that is true, then my OP rant about nanny features stands true.  I would appreciate a guiding philosophy that favored users as competent, rather than incompetent buffoons who need protection from themselves via their software overlords.  Said overlords with their smug righteousness seem to be overly present in our modern day culture and frankly it is insulting.

If I want to accept the risk, that is my choice.  Why should I and all other users be relegated to idiot automatons, incapable of independent decision making?  It is time to lose the bubble wrap and drop the ridiculous guardrails.

Failure provides an excellent path to success.  How about removing those impediments and let people succeed as the individuals that they are?

p.s. thanks queen for your input on this, seriously. my rant is of course is not towards any individual, but rather directed at the flawed ideology in trying to keep people safe from themselves without their consent.  Cheers!

tl;dr
put in an OFF switch for this nanny feature and change the fallacious assumption that people need protection without asking their consent

**Reply by aidasofialily-cmd** [reply]
Exit this sudo mode in GitHub. I these exiting sudo mode [sudo mode](https://help.github.com/articles/sudo-mode/).

**Comment by ekanemworks** [comment]
This is a very annoying feature... I'm trying to add a collaborator and its redirecting me to the sudo mode blog... I haven't been able to do any work

**Comment by LamGC** [comment]
I absolutely support this feature. I hope that Sudo mode should be as short as possible and can specify a validity period during authorization. Even when logging in through a secure key, it is allowed not to enter Sudo mode to ensure minimal session permissions and facilitate temporary work on untrusted devices.

**Comment by DiggyDiggyGraves** [comment]
> GitHub has a so-called [sudo mode](https://help.github.com/articles/sudo-mode/) which I think is a great security feature. However, after entering my password the sudo mode persists for quite a while: (emphasis added)
> 
> > After you’ve performed a sudo-protected action, you’ll only be asked to re-authenticate again after **a few hours** of inactivity.
> 
> How can I manually _exit_ that mode?

![ic_launcher](https://github.com/user-attachments/assets/51535b1f-d0bc-4aa2-b8e1-743965186a00)

**Comment by aidasofialily-cmd** [comment]
Hello @DiggyDiggyGraves, I exit this sudo mode in GitHub. [sudo mode](https://help.github.com/articles/sudo-mode/)

**Comment by aidasofialily-cmd** [comment]
How I cancel sudo mode?

> After you’ve performed a sudo-protected action, you’ll only be asked to re-authenticate again after a few hours of inactivity.

**Comment by aidasofialily-cmd** [comment]
I want to cancel sudo mode? Can you cancel sudo mode? 

> Tip: You are entering [sudo mode](https://docs.github.com/articles/sudo-mode). After you've performed a sudo-protected action, you'll only be asked to re-authenticate again after a few hours of inactivity.

**Comment by ThePikachuN** [comment]
ok i was trying to add my github.com account to my GitHub Desktop and it told me to use Sudo mode when its a verified GitHub application, please try to improve the GitHub sudo mode, seperating a verified application from a unverified application.


<tried using Node.js to add a redirect script but Node.js failed>
